### PR TITLE
Set up the query result cache

### PR DESCRIPTION
**Pull request by rjmac**

### CODE REVIEW

File: query-coordinator/docker/query-coordinator.conf.j2 [hunk 1]
```diff
@@ -11,6 +11,19 @@ com.socrata.query-coordinator {
 
   query-timeout = {{ QUERY_TIMEOUT }}
 
+  {% if CACHE_DATABASE_HOST is defined AND CACHE_DATABASE_POST is defined and CACHE_DATABASE_PASSWORD is defined and CACHE_DATABASE_USERNAME is defined -%}
```
Comment (spaceballone) [inline query-coordinator/docker/query-coordinator.conf.j2]: I think that there's a typo "POST" -> "PORT"

File: query-coordinator/src/main/scala/com/socrata/querycoordinator/caching/cache/config/CacheConfig.scala [hunk 2]
```diff
@@ -10,6 +10,7 @@ class CacheConfig(config: Config, root: String) extends ConfigClass(config, root
     optionally(getString("type")) match {
       case Some("filesystem") => new FilesystemCacheConfig(config, root)
       case Some("postgresql") => new PostgresqlCacheConfig(config, root)
+      case Some("noop") => NoopCacheConfig
```
Comment (spaceballone) [inline query-coordinator/src/main/scala/com/socrata/querycoordinator/caching/cache/config/CacheConfig.scala]: Do you still need this with the conditional logic in the conf.j2?

Comment (rjmac) [inline query-coordinator/src/main/scala/com/socrata/querycoordinator/caching/cache/config/CacheConfig.scala]: Nope, but having a way to explicitly say "no cache" is a good thing anyway.

File: query-coordinator/docker/query-coordinator.conf.j2 (hunk 1 above)
Comment (rjmac) [inline query-coordinator/docker/query-coordinator.conf.j2]: Yep, fixed

**Comment by spaceballone** [comment]
:shipit:

**Pull request merged**